.github/workflows/build-and-publish-on-push.yaml

```diff
@@ -17,34 +16,36 @@ jobs:
         uses: actions/checkout@v2
 
       - name: Install .NET Core SDK
-        uses: actions/setup-dotnet@v1
+        uses: actions/setup-dotnet@v1.7.2
         with:
-          dotnet-version: '3.1.201' # SDK Version to use.
+          dotnet-version: '3.1.x' # use latest 3.1 SDK
+        env:
+          DOTNET_NOLOGO : true
+          DOTNET_CLI_TELEMETRY_OPTOUT : true
 
       - name: Restore required NuGet packages
         run: dotnet restore
         working-directory: src
 
       - name: Build
-        run: dotnet build -c:Release -p:ContinuousIntegrationBuild=true -p:PreReleaseSuffix=.$GITHUB_RUN_NUMBER
+        run: dotnet build -c:Release -p:ContinuousIntegrationBuild=true -p:PreReleaseSuffix=-$GITHUB_RUN_NUMBER
```

Comment:
Maybe adding `--no-restore` could speed this up?

Reply:
```suggestion
        run: dotnet build -c:Release -p:ContinuousIntegrationBuild=true -p:PreReleaseSuffix=-$GITHUB_RUN_NUMBER --no-restore
```

Reply:
Probably not really, but it can't hurt ;)